File: src/test/java/com/frightanic/smn/StationMapTest.java
package com.frightanic.smn;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.nullValue;
import static org.junit.Assert.assertThat;

/**
 * Tests StationMap.
 */
public class StationMapTest {

  /**
   * See method name.
   */
  @Test
  public void shouldReturnNullForUnknownCode() {
    // given
    // when
    // then
    assertThat(StationMap.get("fooBar4711"), is(nullValue()));
  }

  /**
   * See method name.
   */
  @Test
  public void shouldReturnNullForLowerCaseCodes() {
    // given
    // when
    // then
    assertThat(StationMap.get("hoe"), is(nullValue()));
  }

  /**
   * See method name.
   */
  @Test
  public void shouldReturnStationForValidUpperCaseCode() {
    // given
    // when
    // then
    assertThat(StationMap.get("HOE").getCode(), is("HOE"));
  }
}

File: src/test/kotlin/com/frightanic/smn/StationMapTest.kt
package com.frightanic.smn

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.nullValue
import org.junit.jupiter.api.Test

/**
 * Tests StationMap.
 */
internal class StationMapTest {
    @Test
    fun `should return null for unknown code`() {
        // given
        // when
        // then
        assertThat(StationMap["fooBar4711"], `is`(nullValue()))
    }

    @Test
    fun `should return null for lower case codes`() {
        // given
        // when
        // then
        assertThat(StationMap["hoe"], `is`(nullValue()))
    }

    @Test
    fun `should return station for valid upper-case code`() {
        // given
        // when
        // then
        assertThat(StationMap["HOE"]!!.code, `is`("HOE"))
    }
}
